### PR TITLE
SonarQube Generic Execution Test Data reporter

### PR DESCRIPTION
**Pull request by sp-dani-garcia**
Added a `SonarQube` reporter that creates `Generic Execution Test Data` reports

For more information check the following documentation:
- [SonarQube Generic Test Data](https://docs.sonarqube.org/latest/analysis/generic-test/)
- [SonarQube Generic Execution](https://docs.sonarqube.org/latest/analysis/generic-test/#header-2)

### CODE REVIEW

**Comment by horenmar** [comment]
Nice PR, lacking just a simple changes to approval script for normalization across platforms. As it turns out, I cannot push to your PR branch, so this is the diff

```diff
diff --git a/scripts/approvalTests.py b/scripts/approvalTests.py
index 148832f..dad2a96 100755
--- a/scripts/approvalTests.py
+++ b/scripts/approvalTests.py
@@ -29,6 +29,7 @@ filelocParser = re.compile(r'''
 lineNumberParser = re.compile(r' line="[0-9]*"')
 hexParser = re.compile(r'\b(0[xX][0-9a-fA-F]+)\b')
 durationsParser = re.compile(r' time="[0-9]*\.[0-9]*"')
+sonarqubeDurationParser = re.compile(r' duration="[0-9]+"')
 timestampsParser = re.compile(r'\d{4}-\d{2}-\d{2}T\d{2}\:\d{2}\:\d{2}Z')
 versionParser = re.compile(r'Catch v[0-9]+\.[0-9]+\.[0-9]+(-develop\.[0-9]+)?')
 nullParser = re.compile(r'\b(__null|nullptr)\b')
@@ -138,6 +139,7 @@ def filterLine(line, isCompact):
 
     # strip durations and timestamps
     line = durationsParser.sub(' time="{duration}"', line)
+    line = sonarqubeDurationParser.sub(' duration="{duration}"', line)
     line = timestampsParser.sub('{iso8601-timestamp}', line)
     line = specialCaseParser.sub('file:\g<1>', line)
     line = errnoParser.sub('errno', line)
```

**Comment by sp-dani-garcia** [comment]
@horenmar done!

Thank you.

**Comment by JoeyGrajciar** [comment]
@sp-dani-garcia is the xml output accepted  by SonarQube? I see that in strings there is `<` escaped but `>` is not escaped.

**Comment by sp-dani-garcia** [comment]
@JoeyGrajciar yes. We have been using this output with SonarQube in our CI for almost a year now with no issues. Could you give an example please? The code was based in the JUnit reporter and I don't see any special treatment there.

File: projects/SelfTest/Baselines/sonarqube.sw.approved.txt [hunk 1]
```diff
@@ -0,0 +1,1709 @@
...
+      <failure message="REQUIRE(Template_Fixture_2&lt;TestType>::m_a.size() == 1)">
+FAILED:
+	REQUIRE( Template_Fixture_2&lt;TestType>::m_a.size() == 1 )
+with expansion:
+	0 == 1
+Class.tests.cpp:<line number>
+      </failure>
+    </testCase>
+    <testCase name="A TEMPLATE_PRODUCT_TEST_CASE_METHOD based test run that fails - Template_Foo&lt;int>" duration="{duration}">
+      <failure message="REQUIRE(Template_Fixture_2&lt;TestType>::m_a.size() == 1)">
```
[the remark this answers is not in the document]
Comment (sp-dani-garcia) [reply projects/SelfTest/Baselines/sonarqube.sw.approved.txt]: I see. It's the same output that you can see in the JUnit one (https://github.com/catchorg/Catch2/blob/master/projects/SelfTest/Baselines/junit.sw.approved.txt#L104). If there is any problem it should be in the [XmlEncode::encodeTo](https://github.com/catchorg/Catch2/blob/188b3e6511b4c1e8fb3266cf185acf92d825cf96/include/internal/catch_xmlwriter.cpp#L61) method. I don't have the time right now to take a look at it but I think that this problem its out of the scope of this PR.

[the remark this answers is not in the document]
Comment (JoeyGrajciar) [reply projects/SelfTest/Baselines/sonarqube.sw.approved.txt]: Actually it seems okay according to the comment about syntax.

**Comment by horenmar** [comment]
I ended up cherry-picking the commits manually to fix the merge conflict.

**Pull request closed without merging**